### PR TITLE
FLASH-373: Add cpp support for tipb

### PR DESCRIPTION
**Pull request by windtalker**
generate cpp code for tipb

### CODE REVIEW

File: generate-cpp.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+rm -rf proto-cpp && mkdir -p proto-cpp
+rm -rf cpp/tipb && mkdir cpp/tipb
+
+cp proto/* proto-cpp/
+
+function sed_inplace()
+{
+        if [ `uname` == "Darwin" ]; then
```
Comment (zz-jason) [inline generate-cpp.sh]: should be indented by 4 spaces.

File: Makefile [hunk 2]
```diff
@@ -8,3 +8,7 @@ rust:
 
 binlog:
 	./generate-binlog.sh
+
+c++:
+	./generate-cpp.sh
+	rm -rf build_cpp && mkdir build_cpp && cd build_cpp && cmake ../cpp && make && cd .. && rm -rf build_cpp
```
Comment (XuHuaiyu) [inline Makefile]: Why not put this line in generate-cpp.sh ?

File: generate-cpp.sh (hunk 1 above)
Comment (windtalker) [reply generate-cpp.sh]: updated.

File: Makefile (hunk 2 above)
Comment (windtalker) [reply Makefile]: done

**Comment by zanmato1984** [comment]
@XuHuaiyu @zz-jason PTAL

**Pull request merged**